rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -521,20 +546,19 @@ def _query(self, method: Callable, call_order: Sequence[WeightedNode], **kwargs:
         """
         for weighted_node in call_order:
             node_info = weighted_node.node_info
-            web3node = self.web3_mapping.get(node_info, None)
-            if web3node is None and node_info.name != self.etherscan_node_name:
-                continue
-
-            if (
-                web3node is not None and
-                method.__name__ in self.methods_that_query_past_data and
-                web3node.is_pruned is True
-            ):
-                continue
+            if node_info.name != self.etherscan_node_name:
+                if (
+                    (web3node := self.web3_mapping.get(node_info)) is None or
+                    (web3node.is_pruned is True and method.__name__ in self.methods_that_query_past_data)  # noqa: E501
+                ):
+                    continue
+
+                web3 = web3node.web3_instance
+            else:
+                web3 = None
 
             try:
-                web3 = web3node.web3_instance if web3node is not None else None
-                result = method(web3, **kwargs)
+                return method(web3, **kwargs)
```

Comment:
The intention here is to avoid checking multiple times `if web3node is not None` and also have the return stamenet earlier